### ISSUE TITLE
adlfs 2022.11.1 breaks dvc[azure]

### ISSUE THREAD

**Issue by friggog** [iterative/dvc#8619]
# Bug Report

## Description

After recent 2022.11.1 update to adlfs, running dvc with azure remote results in the following error:
```
ERROR: URL 'azure://' is supported but requires these missing dependencies: ['adlfs']. To install dvc with those dependencies, run:

	pip install 'dvc[azure]'

See <[https://dvc.org/doc/install>](https://dvc.org/doc/install%3E) for more info.
```
Despite adlfs being installed.

### Reproduce

1. Install `dvc[azure]`
1. Ensure `adlfs==2022.11.1` is installed, this is the latest version so should be installed by default
1. Run any dvc command involving the remote, e.g., `dvc pull`

### Expected

No error

### Environment information

```
dvc[azure]
adlfs==2022.11.1
```

**Comment by pmrowla** [comment]
This is a bug in adlfs 2022.11.1, I'm assuming they will end up needing to pull this release from pypi. Please downgrade to `2022.10.0` as described in the adlfs issue
https://github.com/fsspec/adlfs/issues/373

**Comment by pmrowla** [comment]
This should be resolved now that adlfs 2022.11.2 is released
